src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,58 @@
+package task;
+
+import utilities.DateTime;
+
+/**
+ * A task with a due date.
+ */
+public class Deadline extends Task {
+    /**
+     * The due date or deadline of the given task in string form.
+     */
+    private String by;
```

Comment:
Could perhaps choose a more descriptive name like `byDate` or `dueDate` which could enhance the readability.